### PR TITLE
feat(date-picker): support type="week"

### LINKED ISSUE #4525: 能否在DatePicker的type选项中添加“week”选项
### This function solves the problem (这个功能解决的问题)
平常的需求中，会遇到选择周的情况。所以，请问下能否添加这个需求，非常感谢。
### Expected API (期望的 API)
<n-date-picker type="week" clearable />

### PR DESCRIPTION
**Pull request by suica**
fixes #4525 

## Todos
- [x] finish `type="week"`
- [ ] check disabled style
- [x] check the i18n files for adding new localized text
- [ ] check the enUS doc
- [x] update the changelog
- [ ] clean the code

### CODE REVIEW

File: src/date-picker/src/DatePicker.tsx [hunk 1]
```diff
@@ -387,6 +391,10 @@ export default defineComponent({
           result.push('confirm')
           return result
         }
+        case 'week': {
+          result.push('confirm')
```
Comment (suica) [inline src/date-picker/src/DatePicker.tsx:395]: Maybe we also don't need this

**Comment by Sepush** [comment]
demos for both en-US and zh-CN but you can add it finally

**Comment by Sepush** [comment]
when impl range need disable the date before the start time

**Comment by Sepush** [comment]
in week mode we do not need the hover style for the speific day 
image
the value/the arg for `on-update:value` the day of firt day of week not the click 
also need handle the `default-time` prop

**Comment by suica** [comment]
Since type `date` and `week` share a simlilar outlook, most code could be resused. So i chose to extend `Date.tsx` to make it support an extra `week` mode.
Do you think it is a good idea? Should we copy the code and make a new week panel?

File: src/date-picker/src/DatePicker.tsx (hunk 1 above)
Comment (suica) [reply src/date-picker/src/DatePicker.tsx]: Removed. Now, `week` and `date` share the same case branch.

**Comment by suica** [comment]
> demos for both en-US and zh-CN but you can add it finally

Added.



> in week mode we do not need the hover style for the speific day image the value/the arg for `on-update:value` the day of firt day of week not the clicked day also need handle the `default-time` prop

Fixed. Removed hover effect for both `--slightly-covered` and `--heavily-covered` modifiers.

@Sepush And, do you think we need a new dedicated file for `week` panel? Now extending the original date panel seems to be a little bit awkward...

**Comment by Sepush** [comment]
> Since type `date` and `week` share a simlilar outlook, most code could be resused. So i chose to extend `Date.tsx` to make it support an extra `week` mode. Do you think it is a good idea? Should we copy the code and make a new week panel?

Just do it if need to

**Comment by AdamsTsui** [comment]
Good job，Thanks!

**Comment by FlareZh** [comment]
Any progress?

**Comment by Sepush** [comment]
cc @07akioni  need code review

**Comment by suica** [comment]
Hi, @07akioni ! i need your general comment on this implementation to make sure I am on the right track, so I can finish the rest todos :)

**Comment by Sepush** [comment]
Thank you for your contribution. @suica 
would you mind add a week number row in the panel with gray color and a dot `.` like  `26.`
I think we can do it in another PR.

**Comment by deiiviidas** [comment]
also need week days translate for example
near weekFormat, [weekDaysFormat ??]

weekDayLong =>[ 'Monday' , ....]
weekDayShort =>[ 'Mon' , ...]

**Pull request merged**